### PR TITLE
config: fix the issue of parse cmdline options

### PR DESCRIPTION
**Pull request by lifupan**
It's should use string.eq() to match option's
key words exactly instead of using string.starts_with()
for single key options and for "key=value" options it's
bettet to match "key=" instead of "key" with string.starts_with()
which would match wrongly such as passed options "agent.log_vsock"
which would match "agent.log" with string.starts_with()
and trigger parsing issues.

Fixes: #96

Signed-off-by: lifupan <lifupan@gmail.com>

### CODE REVIEW

File: src/agent/src/config.rs [hunk 1]
```diff
@@ -197,6 +197,12 @@ mod tests {
         }
 
         let tests = &[
+            TestData {
```
Comment (jodh-intel) [inline src/agent/src/config.rs]: This doesn't seem to be part of this commit??

File: src/agent/src/config.rs [hunk 2]
```diff
@@ -40,20 +40,20 @@ impl agentConfig {
         let cmdline = fs::read_to_string(file)?;
         let params: Vec<&str> = cmdline.split_ascii_whitespace().collect();
         for param in params.iter() {
-            if param.starts_with(DEBUG_CONSOLE_FLAG) {
+            if param.eq(&DEBUG_CONSOLE_FLAG) {
```
Comment (jodh-intel) [inline src/agent/src/config.rs]: Please can you add an extra test for each of these options to assert that, for example, `agent.devmode` set the `agentConfig` variable, but `agent.devmodefoo` doesn't.

Also, I wonder if we should make it clearer which options are simply boolean flags and which take a value. For those that take a value we could change them to be an `_OPTION` [1]:


```rust
const LOG_LEVEL_OPTION: &'static str = "agent.log";
const HOTPLUG_TIMEOUT_OPTION: &'static str = "agent.hotplug_timeout";
```

---
[1] - that would also allow us to update the existing typo nobody had noticed (`HOTPLUG_TIMOUT_FLAG` should be `HOTPLUG_TIMEOUT_FLAG` :smile:).

File: src/agent/src/config.rs (hunk 1 above)
Comment (lifupan) [reply src/agent/src/config.rs]: Hi @jodh-intel , it is. 
It's used to test whether the options of "key=value" and "key${appendix}=value" can be parsed
correctly.

File: src/agent/src/config.rs (hunk 2 above)
Comment (lifupan) [reply src/agent/src/config.rs]: Hi @jodh-intel 
Updated!
